saleor/plugins/webhook/plugin.py

```diff
@@ -417,3 +426,71 @@ def process_payment(
             previous_value,
             **kwargs,
         )
+
+    def __send_shipping_webhook_sync(
+        self,
+        event_type: str,
+        previous_value: List[ExcludedShippingMethod],
+        payload: str,
+    ) -> List[ExcludedShippingMethod]:
+        excluded_methods_map = defaultdict(list)
+
+        # Gather responses from webhooks
+        webhooks = _get_webhooks_for_event(event_type)
+        for webhook in webhooks:
+            response_data = send_webhook_request_sync(
+                webhook.app.name,
+                webhook.target_url,
+                webhook.secret_key,
+                event_type,
+                payload,
+            )
+            if response_data:
+                for method in parse_excluded_shipping_methods_response(response_data):
+                    excluded_methods_map[method.id].append(method)
+
+        # Gather responses for previous plugins
+        for method in previous_value:
+            excluded_methods_map[method.id].append(method)
+
+        # Return a list of excluded methods, unique by id
+        excluded_methods = []
+        for method_id, methods in excluded_methods_map.items():
+            reasons = [m.reason for m in methods if m.reason]
+            reason = None
+            if reasons:
+                reason = " ".join(reasons)
+            excluded_methods.append(ExcludedShippingMethod(id=method_id, reason=reason))
+        return excluded_methods
+
+    def excluded_shipping_methods_for_order(
+        self,
+        order: "Order",
+        available_shipping_methods: List[ShippingMethod],
+        previous_value: List[ExcludedShippingMethod],
+    ) -> List[ExcludedShippingMethod]:
+        payload = generate_excluded_shipping_methods_for_order_payload(
+            order,
+            available_shipping_methods,
+        )
```

Comment:
@kczan 
We shouldn't generate payload if there is no active exclude shipping method webhook.

Reply:
We shoud alse add tracing fot that. 
`import opentracing`
```suggestion
        with opentracing.global_tracer().start_active_span("generate_excluded_shipping_methods_for_order_payload"):
            payload = generate_excluded_shipping_methods_for_order_payload(
                order,
                available_shipping_methods,
            )
```